resharper/src/resharper-unity/Daemon/Stages/Highlightings/UnityNullCoalescingWarning.cs

```diff
@@ -0,0 +1,41 @@
+﻿using JetBrains.DocumentModel;
+using JetBrains.ReSharper.Feature.Services.Daemon;
+using JetBrains.ReSharper.Plugins.Unity.Daemon.Stages.Highlightings;
+using JetBrains.ReSharper.Psi.CSharp;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+
+[assembly: RegisterConfigurableSeverity(UnityNullCoalescingWarning.HIGHLIGHTING_ID,
+    null, UnityHighlightingGroupIds.Unity, UnityNullCoalescingWarning.MESSAGE,
+    "Unity Object types can't be compared using null coalescing. Use ternary operator instead.",
+    Severity.ERROR)]
```

Comment:
Should the warnings be errors or warnings?

Reply:
Error is for something that breaks the compilation and Warning is for something that is highly likely to be incorrect.

Reply:
Error should suffice in this case, I believe. Doesn't break compilation, but it always breaks at runtime.

Reply:
I'm pretty sure marking the highlighting as `Error` will actually prevent compilation. Unity doesn't care about it and will just keep compiling but visual studio won't let you build it until you fix the "error".

Reply:
No, ReSharper/Rider highlights don't affect compilation. This should be a warning, as it's still valid code, but doing something you most likely didn't intend.

Reply:
That makes sense.

Reply:
Yep.